src/codeu/chat/server/View.java

```diff
@@ -85,6 +85,69 @@ public ServerInfo getInfo() {
   @Override
   public Message findMessage(Uuid id) { return model.messageById().first(id); }
 
+  @Override
+  public Collection<String> userStatusUpdate(String name, Uuid owner) {
+
+    Collection<ConversationPayload> allConversations = all(model.conversationPayloadById());
+    Collection<String> contributions = new ArrayList<String>();
+
+    final User foundOwner = model.userById().first(owner);
+    final User foundUser = model.userByText().first(name);
+    final Time lastUpdate = foundOwner.UserUpdateMap.get(foundUser.id);
+
+    for(ConversationPayload conversationPayload : allConversations) {
+      Message currentMessage = model.messageById().first(conversationPayload.firstMessage);
+      boolean foundMessage = false;
+      while(currentMessage != null && foundMessage == false) {
+        if(lastUpdate.compareTo(currentMessage.creation) < 0 && currentMessage.author.equals(foundUser.id)) {
+          contributions.add(model.conversationById().first(conversationPayload.id).title);
+          foundMessage = true;
+        }
+        currentMessage = model.messageById().first(currentMessage.next);
+      }
+      if(!contributions.contains(model.conversationById().first(conversationPayload.id).title)) {
```

Comment:
Is this check necessary? If a user has both created and contributed to a new conversation, you'd want both in the update, right?